x/lockup/keeper/migration.go

```diff
@@ -0,0 +1,85 @@
+package keeper
+
+import (
+	"strconv"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/osmosis-labs/osmosis/x/lockup/types"
+)
+
+var (
+	HourDuration, _    = time.ParseDuration("1h")
+	DayDuration, _     = time.ParseDuration("24h")
+	WeekDuration, _    = time.ParseDuration("168h")
+	TwoWeekDuration, _ = time.ParseDuration("336h")
+	BaselineDurations  = []time.Duration{DayDuration, WeekDuration, TwoWeekDuration}
+)
+
+func normalizeDuration(baselineDurations []time.Duration, allowedDiff time.Duration, duration time.Duration) time.Duration {
+	for _, base := range baselineDurations {
+		if base <= duration && duration < base+allowedDiff {
+			return base
+		}
+	}
+	return duration
+}
+
+// between duration windows of baselineDurations.map((duration) => (duration, duration+durationDiff)).
+func MergeLockupsForSimilarDurations(
+	ctx sdk.Context,
+	k Keeper,
+	ak types.AccountKeeper,
+	baselineDurations []time.Duration,
+	durationDiff time.Duration,
+) {
+	for _, acc := range ak.GetAllAccounts(ctx) {
+		addr := acc.GetAddress()
+		normals := make(map[string]uint64)
+		for _, lock := range k.GetAccountPeriodLocks(ctx, addr) {
+			// ignore multilocks
+			if len(lock.Coins) > 1 {
+				continue
+			}
+			// ignore unlocking locks; they will be removed from the state anyway
+			if lock.IsUnlocking() {
+				continue
+			}
+			coin := lock.Coins[0]
+			normalizedDuration := normalizeDuration(baselineDurations, durationDiff, lock.Duration)
+			key := addr.String() + "/" + coin.Denom + "/" + strconv.FormatInt(int64(normalizedDuration), 10)
```

Comment:
What is this key? Is there a standard function for it?

Reply:
key is just a triplet for (address, denom, duration). We can do
```suggestion
			key := strings.Join([]string{addr.String(), coin.Denom, strconv.FormatInt(int64(normalizedDuration), 10)}, "/")
```

Reply:
Oh its a key just for all the normalized locks, we don't need to parse it ever / it doesn't affect state.

Reply:
SGTM! Latter SGTM